### ISSUE TITLE
[Future deprecation] SP: Using a non-tuple sequence for multidimensional indexing is deprecated and will be changed in pytorch 2.9

### ISSUE THREAD

**Issue by piyifan123** [ByteDance-Seed/VeOmni#267]
When running SP, we observe the following future deprecation warning for torch 2.9. 

https://github.com/ByteDance-Seed/VeOmni/blob/main/veomni/distributed/sequence_parallel/data.py#L64-L86

```
.../Open-VeOmni/veomni/distributed/sequence_parallel/data.py:86: UserWarning: Using a non-tuple sequence for multidimensional indexing is deprecated and will be changed in pytorch 2.9; use x[tuple(seq)] instead of x[seq]. In pytorch 2.9 this will be interpreted as tensor index, x[torch.tensor(seq)], which will result either in an error or a different result (Triggered internally at /pytorch/torch/csrc/autograd/python_variable_indexing.cpp:306.)
  return x[slc].contiguous()
```

**Comment by Luosuu** [comment]
Shall we upgrade to torch 2.9 now?
